File: Shared/Preferences Controllers/ClassicEditorPreferenceController.m
//
//  ClassicEditorPreferenceController.m
//  PPMacho
//
//  Created by C.W. Betts on 7/26/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "ClassicEditorPreferenceController.h"
#import "UserDefaultKeys.h"
#ifdef PLAYERPRO6
#import "PlayerPRO_6-Swift.h"
#else
#import "PlayerPRO_Player-Swift.h"
#endif

typedef struct _tagCoupling {
	int amount;
	NSInteger tag;
} TagCoupling;

static const TagCoupling TrackHeightCoupling[] = {{100, 1}, {110, 2}, {120, 3}, {130, 4}, {140, 5},
	{160, 6}, {180, 7}, {200, 8}};

@implementation ClassicEditorPreferenceController
@synthesize markersCheck;
@synthesize markersLoopValue;
@synthesize markersOffsetValue;
@synthesize notesLengthCheck;
@synthesize tempoNumberValue;
@synthesize tempoUnitValue;
@synthesize trackHeightButton;

+ (instancetype)newPreferenceView
{
	return [[self alloc] init];
}

-(id)init
{
	if (self = [super initWithNibName:@"ClassicPrefs" bundle:nil]) {
		[self setTitle:NSLocalizedStringFromTable(@"Classic Editor", @"PreferenceNames", @"Classic Editor")];
	}
	return self;
}

- (IBAction)changeTrackHeight:(id)sender
{
	int toSet = 0;
	NSInteger tag = [sender tag];
	int i = 0;
	size_t sizeofCoupling = sizeof(TrackHeightCoupling) / sizeof(TrackHeightCoupling[0]);
	for (i = 0; i < sizeofCoupling; i++) {
		if (TrackHeightCoupling[i].tag == tag) {
			toSet = TrackHeightCoupling[i].amount;
			break;
		}
	}
	if (toSet == 0) {
		toSet = 130;
	}
	[[NSUserDefaults standardUserDefaults] setInteger:toSet forKey:PPCETrackHeight];
	[[NSNotificationCenter defaultCenter] postNotificationName:PPClassicalEditorPreferencesDidChange object:self];
}

- (IBAction)toggleNoteLength:(id)sender
{
	[[NSUserDefaults standardUserDefaults] setBool:[notesLengthCheck state] forKey:PPCEShowNotesLen];
	[[NSNotificationCenter defaultCenter] postNotificationName:PPClassicalEditorPreferencesDidChange object:self];

}

- (IBAction)toggleMarkers:(id)sender
{
	NSInteger state = [markersCheck state];
	[[NSUserDefaults standardUserDefaults] setBool:state forKey:PPCEShowMarkers];
	[markersLoopValue setEnabled:state];
	[markersOffsetValue setEnabled:state];
	[[NSNotificationCenter defaultCenter] postNotificationName:PPClassicalEditorPreferencesDidChange object:self];
}

- (void)awakeFromNib
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	{
		NSInteger theRate = [defaults integerForKey:PPCETrackHeight];
		NSInteger toSet = 0;
		int i = 0;
		size_t sizeofCoupling = sizeof(TrackHeightCoupling) / sizeof(TagCoupling);
		for (i = 0; i < sizeofCoupling; i++) {
			if (TrackHeightCoupling[i].amount == theRate) {
				toSet = TrackHeightCoupling[i].tag;
				break;
			}
		}
		if (toSet == 0) {
			toSet = 4;
		}
		[trackHeightButton selectItemAtIndex:toSet - 1];
	}

	[tempoUnitValue setIntegerValue:[defaults integerForKey:PPCETempoUnit]];
	[tempoNumberValue setIntegerValue:[defaults integerForKey:PPCETempoNum]];
	[markersLoopValue setIntegerValue:[defaults integerForKey:PPCEMarkerLoop]];
	[markersOffsetValue setIntegerValue:[defaults integerForKey:PPCEMarkerOffset]];
	[notesLengthCheck setState:[defaults boolForKey:PPCEShowNotesLen]];
	
	BOOL markersVal = [defaults boolForKey:PPCEShowMarkers];
	[markersLoopValue setEnabled:markersVal];
	[markersOffsetValue setEnabled:markersVal];
	[markersCheck setState:markersVal];
}

@end

File: Shared/Preferences Controllers/ClassicEditorPreferenceController.swift
//
//  ClassicEditorPreferenceController.swift
//  PPMacho
//
//  Created by C.W. Betts on 8/12/14.
//
//

import Cocoa

let TrackHeightCoupling = [TagCoupling(100, 1), TagCoupling(110, 2), TagCoupling(120, 3), TagCoupling(130, 4), TagCoupling(140, 5),
	TagCoupling(160, 6), TagCoupling(180, 7), TagCoupling(200, 8)];

class ClassicEditorPreferenceController: NSViewController, PPPreferenceObject {
	@IBOutlet var tempoNumberValue:		NSTextField!
	@IBOutlet var markersLoopValue:		NSTextField!
	@IBOutlet var markersOffsetValue:	NSTextField!
	@IBOutlet var markersCheck:			NSButton!
	@IBOutlet var notesLengthCheck:		NSButton!
	@IBOutlet var trackHeightButton:	NSPopUpButton!
	@IBOutlet var tempoUnitValue:		NSTextField!
	
	class func newPreferenceView() -> Self {
		var nullBundle: NSBundle! = nil
		var ourself = self(nibName: "ClassicPrefs", bundle: nullBundle)
		
		var aTitle = NSLocalizedString("Classic Editor", tableName:"PreferenceNames", comment: "Classic Editor");
		ourself.title = aTitle
		return ourself
	}
	
	required init(coder: NSCoder!) {
		super.init(coder: coder)
	}
	
	required override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
		super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
	}
	
	override func awakeFromNib() {
		super.awakeFromNib()
		// Do view setup here.
		let defaults = NSUserDefaults.standardUserDefaults()
		
		let aRate = Int32(defaults.integerForKey(PPCETrackHeight))
		
		var toSet = 0;
		for iaa in TrackHeightCoupling {
			if aRate == iaa.amount {
				toSet = iaa.tag;
				break;
			}
		}
		if (toSet == 0) {
			toSet = 4;
		}
		trackHeightButton.selectItemAtIndex(toSet - 1)
		
		tempoUnitValue.integerValue = defaults.integerForKey(PPCETempoUnit)
		markersLoopValue.integerValue = defaults.integerForKey(PPCEMarkerLoop)
		markersOffsetValue.integerValue = defaults.integerForKey(PPCEMarkerOffset)
		notesLengthCheck.state = defaults.boolForKey(PPCEShowNotesLen) ? NSOnState : NSOffState
		
		let markersVal = defaults.boolForKey(PPCEShowMarkers)
		markersLoopValue.enabled = markersVal
		markersOffsetValue.enabled = markersVal
		markersCheck.state = markersVal ? NSOnState : NSOffState
	}
	
	@IBAction func changeTrackHeight(sender: AnyObject!) {
		
	}
	
	@IBAction func toggleNoteLength(sender: AnyObject!) {
		
	}
	
	@IBAction func toggleMarkers(sender: AnyObject!) {
		
	}

}
